elastic/integrations
# [System] Add journald support for auth and syslog data streams

## Note for reviewers

I did my best to get events that would test all the grok expressions, I mostly tried to get similar messages to the `syslog` and `auth` dataset. The `auth` dataset might be missing some from PAM because I wasn't sure how to generate them.


## Proposed commit message

This commit introduces support for Journald in the system integration,
adding a "journald" input for both auth and syslog data.

Conditions are used to select when to run the log input or the journal input, it
defaults to running the Journald input on Debian 12 and Amazon Linux 2023.

The minimum stack version is updated to `8.17.0` so the integration can use the
facilities filtering option from the journald input added in `8.16.0` and some extra fields
from the host provider added in `8.17.0`.

## Checklist


## Author's Checklist

- [x] Ensure `data_stream.dataset` and `event.dataset` match their log input counterpart 
- [x] Wait for https://github.com/elastic/elastic-agent/pull/5941 before merging/releasing this integration version.

## How to test this PR locally
- Build the Elastic-Agent from `main`, you need the changes from (https://github.com/elastic/elastic-agent/pull/5941)
- Create an Elastic-Agent Policy
- Deploy the policy to your hosts.
- If the host is Debian 12 or Amazon Linux 2023, Journald inputs should be used for the auth and syslog data. Any other Linux host should use the log input.

## Related issues
- Closes https://github.com/elastic/integrations/issues/10797 

## Screenshots
<details><summary>Integration configuration</summary>
<p>

![2024-11-22_13-08](https://github.com/user-attachments/assets/5b631337-629b-480d-9941-2c6a1342cbe6)


</p>
</details> 

<details><summary>Dashboards with data from Journald (Debain 12)</summary>
<p>

![2024-11-05_17-53](https://github.com/user-attachments/assets/63053f63-1ca5-4687-9208-a4baea8b94bc)
![2024-11-05_17-54](https://github.com/user-attachments/assets/12ea2aec-f296-4692-bc0c-b0486703ef93)
![2024-11-05_17-54_1](https://github.com/user-attachments/assets/02e7dc7e-c704-45a2-a071-4f337a1dcbe5)
![2024-11-05_17-55](https://github.com/user-attachments/assets/4dea3f64-1bd0-46ee-9d51-f080a80b48f2)

</p>
</details> 

<details><summary>Journald x Log input</summary>
<p>

I enabled both, the journald and the log input for the auth and syslog datasets on a fresh Debian 11 VM, both inputs collected the same amount of events. On the left is the journald input and on the right the log input.

![2024-11-06_17-30](https://github.com/user-attachments/assets/1158672b-99a8-4184-841b-89cf84e9db3f)
![2024-11-06_17-30_1](https://github.com/user-attachments/assets/bf667750-6c8b-4ec9-af03-301e3d82d422)


</p>
</details>
